From tbm@cyrius.com Mon Apr  2 04:44:19 2007
Return-path: <pkg-voip-maintainers-bounces+msp=debian.org@lists.alioth.debian.org>
Envelope-to: debian@purcell.id.au
Delivery-date: Sun, 01 Apr 2007 19:58:39 +0100
Received: from master.debian.org ([70.103.162.29])
	by bristol.purcell.id.au with esmtp (Exim 4.63)
	(envelope-from <pkg-voip-maintainers-bounces+msp=debian.org@lists.alioth.debian.org>)
	id 1HY5G4-0008ND-1W
	for debian@purcell.id.au; Sun, 01 Apr 2007 19:58:39 +0100
Received: from alioth.debian.org ([217.196.43.134])
	by master.debian.org with esmtp (Exim 4.50)
	id 1HY5G2-0002Zc-Hy
	for debian@purcell.id.au; Sun, 01 Apr 2007 18:58:30 +0000
Received: from localhost ([127.0.0.1] helo=alioth.debian.org)
	by alioth.debian.org with esmtp (Exim 4.50)
	id 1HY5G1-0004qK-Tw
	for msp@debian.org; Sun, 01 Apr 2007 18:58:29 +0000
Received: from master.debian.org ([70.103.162.29])
	by alioth.debian.org with esmtp (Exim 4.50) id 1HY5Fu-0004q1-FM
	for pkg-voip-maintainers@lists.alioth.debian.org;
	Sun, 01 Apr 2007 18:58:27 +0000
Received: from qa by master.debian.org with local (Exim 4.50)
	id 1HY5Fs-0002Vm-Sb; Sun, 01 Apr 2007 18:58:20 +0000
Received: from rietz.debian.org ([140.211.166.43]) by master.debian.org
	with esmtp (Exim 4.50) id 1HY57A-0000XJ-5R for
	exiv2@packages.qa.debian.org; Sun, 01 Apr 2007 18:49:20 +0000
Received: from debbugs by rietz.debian.org with local (Exim 4.50) id
	1HY53D-0002cQ-FA; Sun, 01 Apr 2007 18:45:15 +0000
X-Loop: owner@bugs.debian.org
Resent-From: Martin Michlmayr <tbm@cyrius.com>
Resent-To: debian-bugs-dist@lists.debian.org
Resent-CC: Debian KDE Extras Team <pkg-kde-extras@lists.alioth.debian.org>
Resent-Date: Sun, 01 Apr 2007 18:45:12 +0000
Resent-Message-Id: <handler.417176.B.11754530725947@bugs.debian.org>
X-Debian-PR-Message: report 417176
X-Debian-PR-Package: exiv2
X-Debian-PR-Keywords: patch
X-Debian-PR-Source: exiv2
Received: via spool by submit@bugs.debian.org id=B.11754530725947 (code B
	ref -1); Sun, 01 Apr 2007 18:45:12 +0000
Received: (at submit) by bugs.debian.org; 1 Apr 2007 18:44:32 +0000
Received: from sorrow.cyrius.com ([65.19.161.204]) by rietz.debian.org
	with esmtp (Exim 4.50) id 1HY52W-0001WW-5N for submit@bugs.debian.org; 
	Sun, 01 Apr 2007 18:44:32 +0000
Received: by sorrow.cyrius.com (Postfix, from userid 10) id C1C0B64D3E;
	Sun,  1 Apr 2007 18:44:31 +0000 (UTC)
Received: by deprecation.cyrius.com (Postfix, from userid 1000) id
	4EFDD54373; Sun,  1 Apr 2007 19:44:19 +0100 (BST)
Date: Sun, 1 Apr 2007 19:44:19 +0100
From: Martin Michlmayr <tbm@cyrius.com>
To: submit@bugs.debian.org
Message-Id: <20070401184419.GS29178@deprecation.cyrius.com>
References: <E1HXkHy-0006Bs-QR@coconut0>
MIME-Version: 1.0
Content-Type: text/plain;
  charset=us-ascii
Content-Disposition: inline
In-Reply-To: <E1HXkHy-0006Bs-QR@coconut0>
User-Agent: Mutt/1.5.13 (2006-08-11)
Delivered-To: submit@bugs.debian.org
Resent-Sender: Debian BTS <debbugs@bugs.debian.org>
Resent-Date: Sun, 01 Apr 2007 18:45:15 +0000
Delivered-To: exiv2@packages.qa.debian.org
Precedence: list
X-Loop: exiv2@packages.qa.debian.org
X-PTS-Package: exiv2
X-PTS-Keyword: bts
X-Unsubscribe: echo 'unsubscribe exiv2' | mail pts@qa.debian.org
X-BeenThere: pkg-voip-maintainers@lists.alioth.debian.org
X-Mailman-Version: 2.1.5
Reply-To: Martin Michlmayr <tbm@cyrius.com>,
 417176@bugs.debian.org
List-Id: VoIP package maintainers list
	<pkg-voip-maintainers.lists.alioth.debian.org>
List-Unsubscribe: <http://lists.alioth.debian.org/mailman/listinfo/pkg-voip-maintainers>,
	<mailto:pkg-voip-maintainers-request@lists.alioth.debian.org?subject=unsubscribe>
List-Archive: <http://lists.alioth.debian.org/pipermail/pkg-voip-maintainers>
List-Post: <mailto:pkg-voip-maintainers@lists.alioth.debian.org>
List-Help: <mailto:pkg-voip-maintainers-request@lists.alioth.debian.org?subject=help>
List-Subscribe: <http://lists.alioth.debian.org/mailman/listinfo/pkg-voip-maintainers>,
	<mailto:pkg-voip-maintainers-request@lists.alioth.debian.org?subject=subscribe>
Sender: pkg-voip-maintainers-bounces+msp=debian.org@lists.alioth.debian.org
Errors-To: pkg-voip-maintainers-bounces+msp=debian.org@lists.alioth.debian.org
X-SA-Exim-Connect-IP: 70.103.162.29
X-SA-Exim-Rcpt-To: debian@purcell.id.au
X-SA-Exim-Mail-From: pkg-voip-maintainers-bounces+msp=debian.org@lists.alioth.debian.org
X-Spam-Checker-Version: SpamAssassin 3.1.7-deb (2006-10-05) on debian
X-Spam-Level: 
X-Spam-Status: No, score=-2.6 required=1.0 tests=AWL,BAYES_00,
	UNPARSEABLE_RELAY autolearn=ham version=3.1.7-deb
X-Spam-Report: 
	*  0.0 UNPARSEABLE_RELAY Informational: message has unparseable relay
	*      lines
	* -2.6 BAYES_00 BODY: Bayesian spam probability is 0 to 1%
	*      [score: 0.0000]
	* -0.0 AWL AWL: From: address is in the auto white-list
Subject: Bug#417176: FTBFS with GCC 4.3: missing #includes
X-SA-Exim-Version: 4.2.1 (built Tue, 09 Jan 2007 17:23:22 +0000)
X-SA-Exim-Scanned: Yes (on bristol.purcell.id.au)
X-Length: 6955
X-KMail-EncryptionState:  
X-KMail-SignatureState:  
X-KMail-MDN-Sent:  
X-UID: 2335
Status: R
X-Status: NC

Package: exiv2
Version: 0.12-1
Usertags: ftbfs-gcc-4.3
Tags: patch

Your package fails to build with GCC 4.3.  Version 4.3 has not been
released yet but I'm building with a snapshot in order to find errors
and give people an advance warning.  In GCC 4.3, the C++ header
dependencies have been cleaned up.  The advantage of this is that
programs will compile faster.  The downside is that you actually
need to directly #include everything you use (but you really should
do this anyway, otherwise your program won't work with any compiler
other than GCC).  Some background of this can be found at
http://gcc.gnu.org/PR28080

You can reproduce this problem with gcc-snapshot from unstable.

> Automatic build of exiv2_0.12-1 on coconut0 by sbuild/ia64 0.49
...
> /usr/lib/gcc-snapshot/lib/gcc/ia64-linux-gnu/4.3.0/../../../../include/c++/4.3.0/bits/basic_string.h:387: warning: conversion to 'long unsigned int' from 'long int' may alter its value
> g++ -g -Wall -O2 -pedantic -Wall -Wundef -Wcast-align -Wconversion -Wpointer-arith -Wformat-security -Wmissing-format-attribute -W -I.  -c -o utils.o utils.cpp
> utils.cpp: In function 'bool Util::strtol(const char*, long int&)':
> utils.cpp:141: error: 'LONG_MAX' was not declared in this scope
> utils.cpp:141: error: 'LONG_MIN' was not declared in this scope

--- src/utils.cpp~	2007-04-01 18:38:25.000000000 +0000
+++ src/utils.cpp	2007-04-01 18:38:33.000000000 +0000
@@ -49,6 +49,7 @@
 # include <unistd.h>                     // for getopt(), stat()
 #endif
 
+#include <climits>
 #include <cerrno>
 #include <cstdlib>
 #include <cstring>

-- 
Martin Michlmayr
http://www.cyrius.com/



_______________________________________________
Pkg-voip-maintainers mailing list
Pkg-voip-maintainers@lists.alioth.debian.org
http://lists.alioth.debian.org/mailman/listinfo/pkg-voip-maintainers

